clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -1287,6 +1287,16 @@ public void testCommitAsyncNegativeOffset() {
         assertTrue(mockOffsetCommitCallback.exception instanceof IllegalArgumentException);
     }
 
+    @Test(expected = CommitFailedException.class)
+    public void testCommitOffsetSyncCallbackWithCommitFailedExceptionForTimeout() {
```

Comment:
@ijuma , thanks for that because that made me to read and understand the code of `MockClient`. So at this point should I revert the changes of `MockClient` for this PR? Please  let me know your thoughts.